modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/ToolbarDrawerToggleTest.ts

```diff
@@ -68,4 +69,43 @@ describe('browser.tinymce.themes.silver.editor.toolbar.ToolbarDrawerToggleTest',
       pTestToggle({ toolbar: [ 'undo redo', 'bold italic' ], inline: true }, false)
     );
   });
+
+  const pTestEvent = async (toolbarMode: ToolbarMode, command: (editor: Editor) => void) => {
+    const editor = await McEditor.pFromSettings<Editor>({
+      menubar: false,
+      statusbar: false,
+      width: 200,
+      toolbar_mode: toolbarMode,
+      base_url: '/project/tinymce/js/tinymce'
+    });
+    editor.focus();
+
+    const lastState = Singleton.value<boolean>();
+    lastState.set(false);
+    await UiUtils.pWaitForEditorToRender();
+
+    editor.on('ToggleToolbarDrawer', (options: { isToolbarDrawerToggled: boolean }) => {
+      lastState.set(options.isToolbarDrawerToggled);
+    });
+
+    assert.isFalse(lastState.get().getOrDie('Should be set'));
+    command(editor);
+    await Waiter.pWait(10);
```

Comment:
What are we waiting for here? Can it be more precise? Can we wait for a predicate?

Reply:
For some reason the state was not toggled instantly, not sure why. Updated [here](https://github.com/tinymce/tinymce/pull/8238/commits/6a399604a6153909c68b5b2f28a8396b4aa37990) with waiting for the floating toolbar to appear. Hoping this is a bit more clear.